### PR TITLE
ob status, example of pi receiving

### PR DESCRIPTION
**Pull request by AJDurant**
I can't test this, it may well not work. But it should. Obviously when deployed the ob.conf file will need to actually be made with the right values.

I've also added ob-pi.example which you may want stuff from for Roses.

### CODE REVIEW

File: ob.liq [hunk 1]
```diff
@@ -8,9 +11,16 @@ set("log.file.path","/usr/local/var/log/liquidsoap/ob.log")
...
+    system("#{obconnectscript} #{stream} #{status_file} #{email_state}")
+end
+def onOBDisconnect (stream) =
+    system("#{obdisconnectscript} #{stream} #{status_file} #{email_state}")
+end
+
 # Get OB Stream
 
-ob = input.http("http://icecast.ury.york.ac.uk:7070/OB-Line")
+s1 = input.http("http://#{icehost}:#{iceport}/#{obmount}", on_connect = onOBConnect("s1"), on_disconnect = onOBDisconnect("s1"))
```
Comment (LordAro) [inline ob.liq]: s1?

Comment (adfw) [inline ob.liq]: I'm assuming that's meant to be 'stream1', s1 is used for 'studio1' in quite a few places, so should probably be changed if that was the intention.

File: scripts/ob_connect.sh [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+fi
+
+state = $1
+state_file = $2
+email_state = $3
+
+source $state_file
+
+if [ $email_state -eq 1 ]; then
+  echo "Outside Broadcast Source $1 is now available." | mail -s "[RouteOB] Stream Available" "alerts.audioroute@ury.org.uk";
```
Comment (adfw) [inline scripts/ob_connect.sh]: Use $state instead of $1?

File: scripts/ob_disconnect.sh [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+fi
+
+state = $1
+state_file = $2
+email_state = $3
+
+source $state_file
+
+if [ $email_state -eq 1 ]; then
+  echo "Outside Broadcast Source $1 has disconnected." | mail -s "[RouteOB] Stream Lost" "alerts.audioroute@ury.org.uk";
```
Comment (adfw) [inline scripts/ob_disconnect.sh]: Use $state instead of $1?

File: conf/ob.conf.dist [hunk 4]
```diff
@@ -0,0 +1,5 @@
+obmount = 'OB-Line'
+status_file = '/tmp/ob_state.conf'
```
Comment (adfw) [inline conf/ob.conf.dist]: Not really a configuration file, suggest .status or .info (or nothing at all)

File: scripts/ob_connect.sh [hunk 5]
```diff
@@ -0,0 +1,22 @@
+#!/bin/bash
+
+if [ $# -ne 3 ]; then
+    echo "$0: usage: ob_connect.sh <source> <state_file> <email_state>"
+    exit 1
+fi
+
+state = $1
+state_file = $2
+email_state = $3
```
Comment (adfw) [inline scripts/ob_connect.sh]: There should be no spaces between the LHS and RHS (also in ob_disconnect.sh)

File: scripts/ob_connect.sh [hunk 6]
```diff
@@ -0,0 +1,22 @@
...
+
+if [ $email_state -eq 1 ]; then
+  echo "Outside Broadcast Source $1 is now available." | mail -s "[RouteOB] Stream Available" "alerts.audioroute@ury.org.uk";
+fi
+
+declare $1=1;
+
+echo "s1=$s1
+s2=$s2
+s3=$s3" > $state_file
```
Comment (adfw) [inline scripts/ob_connect.sh]: None of these variables have been declared (also in ob_disconnect.sh).

File: scripts/ob_connect.sh [hunk 7]
```diff
@@ -0,0 +1,22 @@
...
+state_file = $2
+email_state = $3
+
+source $state_file
+
+if [ $email_state -eq 1 ]; then
+  echo "Outside Broadcast Source $1 is now available." | mail -s "[RouteOB] Stream Available" "alerts.audioroute@ury.org.uk";
+fi
+
+declare $1=1;
```
Comment (adfw) [inline scripts/ob_connect.sh]: What is this doing? It seems non-obvious, and $1 isn't used again. (See also the same in ob_disconnect except it's $1=0).

File: scripts/ob_connect.sh (hunk 6 above)
Comment (adfw) [inline scripts/ob_connect.sh]: Well, they get declared when the status_file gets run by 'source', but that's not exactly clear.

**Comment by AJDurant** [comment]
These scripts are pretty much unmodified from fs1. The file is that name
and format cause it matches MyRadio.
s1,s2,s3 are source 1,2,3.

The declare does the change to the state variable that should be modified
after they have been loaded in from the existing status.

**Comment by adfw** [comment]
Well, right now this does not work at all. Sample output (of ob_connect.sh) :
````
$./test.sh s1 state.out 1
./test.sh: line 8: state: command not found
./test.sh: line 9: state_file: command not found
./test.sh: line 10: email_state: command not found
./test.sh: line 12: source: filename argument required
source: usage: source filename [arguments]
./test.sh: line 14: [: -eq: unary operator expected
./test.sh: line 22: $state_file: ambiguous redirect
````

**Comment by mstratford** [comment]
Hmm, wondering if this is still required? I doubt it lines up with any of the recent configs for doing Pi broadcasting for roses. @hicks927 @markspolakovs thoughts?

**Comment by AJDurant** [comment]
Highly unlikely to be. I can't remember what prompted the creation of it back in 2017, it was probably something that I had lying around that had got lost and was mentioned at the time.

**Pull request closed without merging**